controlplane/src/core/bufservices/PlatformService.ts

```diff
@@ -4102,29 +4103,60 @@ export default function (opts: RouterOptions): Partial<ServiceImpl<typeof Platfo
 
         const generatedAPIKey = ApiKeyGenerator.generate();
 
-        // check if the user is authorized to perform the action
-        for (const targetId of req.federatedGraphTargetIds) {
-          await opts.authorizer.authorize({
-            db: opts.db,
-            graph: {
-              targetId,
-              targetType: 'federatedGraph',
-            },
-            headers: ctx.requestHeader,
-            authContext,
-          });
-        }
+        const rbac = await orgRepo.getFeature({
+          organizationId: authContext.organizationId,
+          featureId: 'rbac',
+        });
 
-        for (const targetId of req.subgraphTargetIds) {
-          await opts.authorizer.authorize({
-            db: opts.db,
-            graph: {
-              targetId,
-              targetType: 'subgraph',
-            },
-            headers: ctx.requestHeader,
-            authContext,
-          });
+        if (rbac) {
+          // check if the user is authorized to perform the action
+          for (const targetId of req.federatedGraphTargetIds) {
+            await opts.authorizer.authorize({
+              db: opts.db,
+              graph: {
+                targetId,
+                targetType: 'federatedGraph',
+              },
+              headers: ctx.requestHeader,
+              authContext,
+            });
+          }
+
+          for (const targetId of req.subgraphTargetIds) {
+            await opts.authorizer.authorize({
+              db: opts.db,
+              graph: {
+                targetId,
+                targetType: 'subgraph',
+              },
+              headers: ctx.requestHeader,
+              authContext,
+            });
+          }
+
+          if (req.allowAllResources && !authContext.isAdmin) {
```

Comment:
Let's do the two checks before we validate the graphs.

Reply:
Done